velox/functions/lib/MapConcat.cpp

```diff
@@ -118,6 +118,10 @@ class MapConcatFunction : public exec::VectorFunction {
       for (vector_size_t i = 1; i < mapSize; i++) {
         if (combinedKeys->equalValueAt(
                 combinedKeys.get(), mapOffset + i, mapOffset + i - 1)) {
+          if (FLAGS_velox_exception_on_duplicate_map_values) {
```

Comment:
Does it make sense to use a config to control different behaviors? Query config could be accessed via `context.execCtx()->queryCtx()->queryConfig()`.

Reply:
A config would make sense.

>  It will be reused in other map functions as well.

What is the full list of such functions. It would be nice to update all of them together to avoid inconsistent behavior.

Reply:
Below is the list of spark functions which utilizes the config `spark.sql.mapKeyDedupPolicy`

Spark function => Velox function
-----------------------------------------
CreateMap => Map
MapFromArrays => Map
MapFromEntries => MapFromEntries
StringToMap => StringToMap
MapConcat => MapConcat
TransformKeys => TransformKeys

Spark handles the duplicates (exception, last_win) based on config value. So, yes it would make sense to create a new config which can be mapped to spark config in Gluten.

@mbasmanova @rui-mo 
I will address `MapConcat` and `MapFromEntries` in the current PR based on the feedback and rest functions in a separate PR.

Reply:
I see that `Map`, `MapFromEntries` & `TransformKeys` use [checkDuplicateKeys](https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/CheckDuplicateKeys.cpp) which in turn does not allow duplicate values. As per Spark's policy requirement (`LAST_WIN`, `EXCEPTION`), I can create a method say `handleDuplicateKeys()`  and all map functions can call this method to handle duplicates.

Reply:
maybe reuse this method https://github.com/facebookincubator/velox/blob/main/velox/functions/lib/CheckDuplicateKeys.cpp#L20C1-L25C12 ?